### PR TITLE
Tidied up the destroy methods and moved the ticker class

### PR DESCRIPTION
**Pull request by alvinsight**

### CODE REVIEW

**Comment by englercj** [comment]
:+1:

File: src/core/display/Container.js [hunk 1]
```diff
@@ -544,3 +544,12 @@ Container.prototype.renderCanvas = function (renderer)
...
     }
 };
+
+/**
+ * Destroys the container
+ *
+ */
+Container.prototype.destroy = function ()
+{
+    this.children = null;
```
Comment (drkibitz) [inline src/core/display/Container.js]: I might be missing an obvious case here, but I thought I ask anyway ;)

Could be be recursive, or maybe recursive with an arg `Container.prototype.destroy = function (recursive)`, or maybe another method?

At the very least, should this not call `removeChildren`?

Comment (englercj) [inline src/core/display/Container.js]: It should definitely call the destroy method of children, and probably remove them as well.

Comment (GoodBoyDigital) [inline src/core/display/Container.js]: Hey guys. I think we could add a 'destroyChildren' property to the function (much like the destroy texture function) That we we have both angles covered.

Comment (GoodBoyDigital) [inline src/core/display/Container.js]: We should deffo call ```removeChildren``` too. That way we get rid of the parent references from the children.

Comment (englercj) [inline src/core/display/Container.js]: Yeah, a param just like texture would be awesome.

**Pull request merged**